pkg/kubelet/kuberuntime/instrumented_services.go

```diff
@@ -59,11 +62,26 @@ func recordError(operation string, err error) {
 	}
 }
 
+// transformError lets us distinguish an error that was specific to grpc versus actual operation.
+// for example, if the grpc protocol did not fail, but the operation did, errors appear in the form:
+//   rpc error: code = 2 desc = <actual error>
+// if the error was not produced by the rpc system itself, we return an error with just desc
+func transformError(err error) error {
+	if err == nil {
+		return nil
+	}
+	// if we have an actual error, filter out code unknown
+	errorDesc := grpc.ErrorDesc(err)
+	// and give a new event with actually useful information
+	return fmt.Errorf("%v", errorDesc)
+}
+
 func (in instrumentedRuntimeService) Version(apiVersion string) (*runtimeapi.VersionResponse, error) {
 	const operation = "version"
 	defer recordOperation(operation, time.Now())
 
 	out, err := in.service.Version(apiVersion)
+	err = transformError(err)
 	recordError(operation, err)
```

Comment:
Just a nit question, Can recordError, do this first thing in the method so we don't have to make the same change in many lines?